orangecontrib/prototypes/widgets/owimagegrid.py

```diff
@@ -0,0 +1,1135 @@
+            horizontalScrollBarPolicy=Qt.ScrollBarAlwaysOff
+        )
+        self.mainArea.layout().addWidget(self.thumbnailView)
+        self.scene = self.thumbnailView.scene()
+        self.scene.selectionChanged.connect(self.onSelectionChanged)
+        self.loader = ImageLoader(self)
+
+    def process(self, size_x=0, size_y=0):
+        if self.sim_grid:
+            self.sim_grid.process(size_x, size_y)
+
+    def sizeHint(self):
+        return QSize(600, 600)
+
+    # checks the input data for the right meta-attributes and finds the image filename.
+    @Inputs.data
+    def setData(self, data):
+        self.closeContext()
+        self.clear()
+
+        self.data = data
+
+        if data is not None:
+            domain = data.domain
+            self.allAttrs = (domain.class_vars + domain.metas +
+                             domain.attributes)
+
+            self.stringAttrs = [a for a in domain.metas if a.is_string]
+            self.domainAttrs = len(domain.attributes)
+
+            self.stringAttrs = sorted(
+                self.stringAttrs,
+                key=lambda attr: 0 if "type" in attr.attributes else 1
+            )
+
+            indices = [i for i, var in enumerate(self.stringAttrs)
+                       if var.attributes.get("type") == "image"]
+            if indices:
+                self.imageAttr = indices[0]
+
+            self.imageAttrCB.setModel(VariableListModel(self.stringAttrs))
+
+            self.openContext(data)
+
+            self.imageAttr = max(min(self.imageAttr, len(self.stringAttrs) - 1), 0)
+
+            if self.isValidData():
+                self.sim_grid = ImageGrid(data)
+                self.setupScene()
+        else:
+            self.info.setText("Waiting for input.\n")
+
+    def clear(self):
+        self.data = None
+        self.sim_grid = None
+        self.error()
+        self.imageAttrCB.clear()
+        self.clearScene()
+
+    def isValidData(self):
+        return self.data and self.stringAttrs and self.domainAttrs
+
+    # loads the images and places them into the viewing area
+    def setupScene(self, process_grid=True):
+        self.error()
+        if self.data:
+            attr = self.stringAttrs[self.imageAttr]
+            assert self.thumbnailView.count() == 0
+            size = QSizeF(self.imageSize, self.imageSize)
+
+            if process_grid and self.sim_grid:
+                self.process()
+                self.columns = self.sim_grid.size_x
+                self.rows = self.sim_grid.size_y
+
+            self.thumbnailView.setFixedColumnCount(self.columns)
+            self.thumbnailView.setFixedRowCount(self.rows)
+
+            for i, inst in enumerate(self.sim_grid.image_list):
+
+                thumbnail = GraphicsThumbnailWidget(QPixmap(), crop=self.cell_fit == 1)
+                thumbnail.setThumbnailSize(size)
+                thumbnail.instance = inst
+                self.thumbnailView.addThumbnail(thumbnail)
```

Comment:
How about if empty space weren't an item one can click/select?

Reply:
fixed in [8f8c7c9](https://github.com/biolab/orange3-prototypes/pull/101/commits/8f8c7c930f785ef15c4da95d5210cbe40d3a69ad)